File: Leetcode/contains-duplicate.java

import java.util.*;

class Main {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        System.out.println("Size of array : ");
        int n= sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0 ; i < n ; i++){
            arr[i] = sc.nextInt();
        }
        System.out.println(containsDuplicate(arr));
    }

    public static boolean containsDuplicate(int[] nums) {

        Map<Integer,Integer> map = new HashMap<Integer,Integer>();
        for (int i = 0 ; i < nums.length ; i++){
            if(!map.containsKey(nums[i])){
                map.put(nums[i],1);
            }
            else{
                return true;
            }
        }
        return false;
    }
}



// Example test case
// Input: Size = 4
// 1 2 3 1
// Output: true

File: Leetcode/ContainsDuplicate.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

bool DuplicateElement(int* arr,int arr_size);

int main()
{
    int size,i;

    printf("The number of elements in the array: ");
    scanf("%d",&size);

    int* arr = malloc(size*sizeof(int));

    printf("Enter the elements: ");
    for(i=0;i<size;i++)
       scanf("%d",&arr[i]);

    if(DuplicateElement(arr,size) == true)
        printf("True");
    else
        printf("False");

    return 0;
}

bool DuplicateElement(int* arr,int arr_size)
{
    int i,j;
    for(i=0;i<arr_size;i++)
    {
        for(j=i+1;j<arr_size;j++)
        {
            if(arr[i] == arr[j])
               return true;
        }
    }
    return false;
}

/*
Given an integer array nums, return true if any value appears at least
twice in the array, and return false if every element is distinct.
Input: nums = [1,2,3,1]
Output: true
*/
